Styleguide update for installation_distros.rst - AI-assisted (#2855)
ansible/ansible-documentation: docs/docsite/rst/installation_guide/installation_distros.rst

Installing Ansible on specific operating systems
================================================

.. note:: 
   These instructions come from their respective communities. 
   If you encounter bugs or issues, file them with that community to update these instructions. 
   Ansible maintains only the ``pip install`` instructions.

You can always :ref:`install the ansible package from PyPI using pip <intro_installation_guide>` on most systems.
The community also packages and maintains Ansible for various Linux distributions.

This guide shows you how to install Ansible from different distribution package repositories.

Requirements for adding new distributions
-----------------------------------------

Package maintainers who want to add instructions for another distribution to this guide must meet the following requirements:

* Ensure the distribution provides a reasonably up-to-date version of ``ansible``.
* Keep ``ansible-core`` and ``ansible`` versions synchronized to the extent that the build system allows.
* Provide a way to contact the distribution maintainers as part of the instructions. 

>   Distribution maintainers are also encouraged to join the `Ansible Packaging <https://matrix.to/#/#packaging:ansible.com>`_ Matrix room.

Reviewer: ```suggestion
* Provide a way to contact the distribution maintainers as part of the instructions.
* Distribution maintainers are also encouraged to join and monitor the `Ansible Packaging <https://matrix.to/#/#packaging:ansible.com>`_ Matrix room.
```
I would make these points separate. They imo don't overlap much.

.. contents::
  :local:

Installing Ansible on Fedora Linux

> -----------------------------------

Reviewer: ```suggestion
----------------------------------
```

Author: oddly enuf, AI kept adding the extra - there.

Fedora Linux provides both the full Ansible package and the minimal ansible-core package through the standard repositories.

Install the full ``ansible`` package:

.. code-block:: bash

    $ sudo dnf install ansible

Install the minimal ``ansible-core`` package:

.. code-block:: bash

    $ sudo dnf install ansible-core

Fedora repositories include several Ansible collections as standalone packages that you can install alongside ``ansible-core``.
For example, install the ``community.general`` collection:

.. code-block:: bash

   $ sudo dnf install ansible-collection-community-general

See the `Fedora Packages index <https://packages.fedoraproject.org/search?query=ansible-collection>`_
for a complete list of Ansible collections packaged in Fedora.

Contact the package maintainers by `filing a bug <https://bugzilla.redhat.com/enter_bug.cgi>`_ against the ``Fedora`` product in Red Hat Bugzilla.

Installing Ansible from EPEL

> -----------------------------

Reviewer: ```suggestion
----------------------------
```

> The EPEL repository provides Ansible packages for Enterprise Linux distributions.

Reviewer: ```suggestion
```
I'm not an expert, but this line sounds a little weird and unnecessary to me. I would suggest removing it.

> If you use CentOS Stream, Almalinux, Rocky Linux, or related distributions, you can install ``ansible`` or Ansible collections from the community-maintained `EPEL <https://docs.fedoraproject.org/en-US/epel/>`_ (Extra Packages for Enterprise Linux) repository.

Reviewer: ```suggestion
If you use CentOS Stream, Almalinux, Rocky Linux, or related distributions, you can install ``ansible`` or Ansible collections from the community-maintained `EPEL <https://docs.fedoraproject.org/en-US/epel/>`_ (Extra Packages for Enterprise Linux) repository:
```

First, `enable the EPEL repository <https://docs.fedoraproject.org/en-US/epel/#_quickstart>`_.

> Then use the same ``dnf`` commands as for Fedora Linux.

Reviewer: ```suggestion
1. `Enable the EPEL repository <https://docs.fedoraproject.org/en-US/epel/#_quickstart>`_.
2. Use the same ``dnf`` commands as for Fedora Linux.
```

Author: @Andersson007 I'm curious if your AI-assisted prompts would have caught this one and turned it into a procedure?

Reviewer: @samccann heh, tried, nope. Even putting another item like ` * If you encounter paragraphs / a sequence of sentences containing words like "First", "Second", "Then" and alike, consider modifying them to a numbered list. ` doesn't help (Gemini 2.5-pro).

Contact the package maintainers by `filing a bug <https://bugzilla.redhat.com/enter_bug.cgi>`_ against the ``Fedora EPEL`` product in Red Hat Bugzilla.

Installing Ansible on OpenSUSE Tumbleweed/Leap

> -----------------------------------------------

Reviewer: ```suggestion
----------------------------------------------
```

OpenSUSE provides Ansible packages through the standard package manager.

.. code-block:: bash

    $ sudo zypper install ansible

See the `OpenSUSE Support Portal <https://en.opensuse.org/Portal:Support>`_ for additional help with Ansible on OpenSUSE.

.. _from_apt:

Installing Ansible on Ubuntu

> -----------------------------

Reviewer: ```suggestion
----------------------------
```

    $ sudo add-apt-repository --yes --update ppa:ansible/ansible
    $ sudo apt install ansible

.. note:: 
   On older Ubuntu distributions, "software-properties-common" is called "python-software-properties". 
   You may want to use ``apt-get`` rather than ``apt`` in older versions. 
   Also, only newer distributions (18.04, 18.10, and later) have a ``-u`` or ``--update`` flag. 
   Adjust your script as needed.

File any issues in `the PPA's issue tracker <https://github.com/ansible-community/ppa/issues>`_.

Installing Ansible on Debian

> -----------------------------

Reviewer: ```suggestion
----------------------------
```

    - Ubuntu 18.04 (Bionic)
    - ``bionic``

> The following example assumes that you already have wget and gpg installed (``sudo apt install wget gpg``).

Reviewer: ```suggestion
The following example assumes that you already have ``wget`` and ``gpg`` installed.
```
they should figure out ourselves how to install those:)

    $ echo "deb [signed-by=/usr/share/keyrings/ansible-archive-keyring.gpg] http://ppa.launchpad.net/ansible/ansible/ubuntu $UBUNTU_CODENAME main" | sudo tee /etc/apt/sources.list.d/ansible.list
    $ sudo apt update && sudo apt install ansible

.. note::
   The quotation marks around the keyserver URL are important.

>    Around the "echo deb" command, use double quotes rather than single quotes.

Reviewer: ```suggestion
   Use double quotes around the keyserver URL and in the "echo deb" command like in the example above.
```
I see there were double quotes in the original first sentence. If we use just "quotation", the `Around the "echo deb" command, use double quotes rather than single quotes.` sentence will imo imply that it can be single or double in the first sentence, but  the double ones in the second. But I'm not sure if it would work, let's simplify.

These commands download the signing key and add an entry to apt's sources pointing to the PPA.

Previously, you may have used ``apt-key add``.
The ``apt-key add`` approach is now `deprecated <https://askubuntu.com/a/1307181>`_ for security reasons (on Debian, Ubuntu, and elsewhere).

As such, we do NOT add the key to ``/etc/apt/trusted.gpg.d/`` or to ``/etc/apt/trusted.gpg`` where the key would be allowed to sign releases from ANY repository.

Installing Ansible on Arch Linux

> ---------------------------------

Reviewer: ```suggestion
--------------------------------
```
